src/sqlfluff/rules/L003.py

```diff
@@ -408,7 +414,7 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
                 # First non-whitespace element is our trigger
                 memory.trigger = segment
 
-        is_last = self.is_final_segment(context)
+        is_last = context.segment is context.final_segment
```

Comment:
`is_final_segment()` is very complex and slow. Switching to the similar, much faster `RuleContext.final_segment`.

Reply:
Legend

Reply:
Maybe rename the function? The property (`RuleContext.final_segment`) is the actual, final segment. The function supports optional filtering of `meta` segments. Currently, only one rule (L052) uses the function, so we could potentially move it into L052.py. Feel free to create an issue or PR for this. I'll probably go ahead and merge, since this doesn't seem like a blocker.